### PR TITLE
test: Repair and Refactor Requester Pays Tests

### LINKED ISSUE #1792: Repair System Tests To Properly Assert
Currently, assertion failures are quietly handled in `system-tests` as `assert` is handled in a callback in an un-awaited function (nothing to catch it, yet it isn't propagated) https://github.com/googleapis/nodejs-storage/blob/31494570328c9b89dd52f7062072543d48af328b/system-test/storage.ts#L1904-L1914 making debugging broken builds for the following PRs a multi-dev, day+ long effort:
- https://github.com/googleapis/nodejs-storage/pull/1790
- https://github.com/googleapis/nodejs-storage/pull/1783

This should be fixed (and perhaps renamed from `doubleTest`) to improve debuggability (log message and proper throws) and save time (avoid waiting for timeouts for a known, immediate failure and dev time to debug).

### PR DESCRIPTION
**Pull request by danielbankhead**
Fixes #1792, #1794 🦕

### CODE REVIEW

File: system-test/storage.ts [hunk 1]
```diff
@@ -1901,16 +1852,51 @@ describe('storage', () => {
...
+          try {
+            // Should raise an error on requester pays bucket
+            await testFunction({});
+          } catch (e) {
+            expectedError = e;
+          }
+
+          assert(expectedError instanceof Error);
+          assert(
+            expectedError.message.includes(failureMessage),
```
Comment (ddelgrosso1) [inline system-test/storage.ts:1893]: In light of some conversation that was had, is it currently possible to check the code here instead of the message?

Comment (danielbankhead) [reply system-test/storage.ts:1893]: Yea, I wish... it turns out the code isn't returned properly upstream (just like the internal note on the Java client when this message broke all the things...)

Comment (danielbankhead) [reply system-test/storage.ts:1893]: I'll file a bug internally to fix w/ the team tagged, this way we can update the tests with the code once its fixed upstream

File: system-test/storage.ts [hunk 2]
```diff
@@ -1901,16 +1852,51 @@ describe('storage', () => {
...
 
-        function doubleTest(testFunction: Function) {
+        /**
+         * A type for requester pays functions. The `options` parameter
+         *
+         * Using `typeof USER_PROJECT_OPTIONS` ensures the function should
+         * support requester pays options, otherwise will fail early at build-
+         * time rather than runtime.
+         */
+        type requesterPaysFunction<
```
Comment (ddelgrosso1) [inline system-test/storage.ts:1862]: 👍

**Review by ddelgrosso1 — APPROVED** [review]
Looks much better, thanks for handling this.

**Pull request merged**